### PR TITLE
added orders for b2b

### PR DESCRIPTION
**Pull request by vadym-uskov**

### CODE REVIEW

File: tests/api/b2b/glue/order_endpoints/orders/positive.robot [hunk 1]
```diff
@@ -0,0 +1,363 @@
+*** Settings ***
+Suite Setup    SuiteSetup
+Test Setup    TestSetup
+Resource    ../../../../../../resources/common/common_api.robot
+Default Tags    glue
+
+*** Test Cases ***
+#GET requests
```
Comment (OKrivtsova) [inline tests/api/b2b/glue/order_endpoints/orders/positive.robot:8]: - we also need a test with products that have measurement units (e.g. cable) - and check how order shows their amount ([data][attributes][items][0][amount]) and qty and maybe we can include sales units or measurement units them
- we need to show an order that has more than one item with different qtys (they will be split in order into products with qty 1)
- we need is the order with a non-split product (by default is we order 10 or more identical products they will not be split and will shown as 1 item
- we may also need an order in a different mode/currency (e.g. net mode, chf)
- we need order with includes of concrete-products, abstract-products
- we need order pagination checked [https://glue.mysprykershop.com/orders?page[offset]=2&page[limit]=1] - shows the second order only

File: resources/environments/environments_api_b2b.json [hunk 2]
```diff
@@ -112,8 +112,11 @@
         "bundled_product_2_abstract_sku": "M1017905",
         "bundled_product_3_abstract_sku": "M24605",
         "bundled_product_1_concrete_sku": "104939",
+        "bundled_product_1_concrete_name": "Dauphin vertebral disc swivel chair, synchronous mechanism",
         "bundled_product_2_concrete_sku": "574995",
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply resources/environments/environments_api_b2b.json:116]: variables were added only for b2b

File: resources/environments/environments_api_b2b.json [hunk 3]
```diff
@@ -112,8 +112,11 @@
...
+        "bundled_product_2_concrete_name": "UPLINER standing desk, electric - without traverse, WxD 1800 x 800 mm - substructure silver, light gray",
         "bundled_product_3_concrete_sku": "426581",
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply resources/environments/environments_api_b2b.json:118]: variable was added only for b2b

File: resources/environments/environments_api_b2b.json [hunk 4]
```diff
@@ -112,8 +112,11 @@
         "bundled_product_2_abstract_sku": "M1017905",
         "bundled_product_3_abstract_sku": "M24605",
         "bundled_product_1_concrete_sku": "104939",
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply resources/environments/environments_api_b2b.json:114]: variables were added only for b2b

File: tests/api/b2b/glue/order_endpoints/orders/negative.robot [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    ...  AND    Save value to a variable:    [data][attributes][orderReference]    order_id
+    ...  AND    I set Headers:    Authorization=fake_token
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/order_endpoints/orders/negative.robot:17]: fixed

File: tests/api/b2b/glue/order_endpoints/orders/negative.robot [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    ...  AND    Save value to a variable:    [data][attributes][orderReference]    order_id
+    ...  AND    I set Headers:    Authorization=
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/order_endpoints/orders/negative.robot:32]: fixed

[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/order_endpoints/orders/negative.robot:32]: fixed

File: tests/api/b2b/glue/order_endpoints/orders/negative.robot [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...  AND    I set Headers:    Authorization=${token}
+    When I send a GET request:    /orders/fake_order_id
+    Then Response status code should be:    404
+    And Response reason should be:    Not Found
+    And Response should return error message:    "Cant find order by the given order reference"
+    And Response should return error code:    801
+
+Get_order_without_order_id
+# Created the bug related to order_id as required parameter - https://spryker.atlassian.net/browse/CC-16510
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/order_endpoints/orders/negative.robot:49]: - bug ticket deleted
- commented 'Get_order_without_order_id' negative test
- added 'Get_customer_orders_list_without_order_id' positive test for /orders endpoint

File: tests/api/b2b/glue/order_endpoints/orders/positive.robot (hunk 1 above)
Comment (vadym-uskov) [reply tests/api/b2b/glue/order_endpoints/orders/positive.robot:8]: - added 'Get_order_by_order_id_with_sales_unit' test
- added 'Get_order_by_order_id_with_different_items_and_quantity' test
- added 'Get_order_by_order_id_with_item_&_quantity>10' test
- added 'Get_order_by_order_id_with_net_mode_&_chf_currency_&_express_shipment_method' test
- there is no possibility to use 'concrete-products' or 'abstract-products' in /orders/{order_id} endpoint
- added 'Get_customer_orders_list_without_order_id_with_pagination'

File: tests/api/b2b/glue/order_endpoints/orders/positive.robot [hunk 8]
```diff
@@ -0,0 +1,527 @@
...
+    #items
+    And Response body parameter should be:    [data][attributes][items][0][name]    ${concrete_product_random_weight_name}
+    And Response body parameter should be:    [data][attributes][items][0][sku]    ${concrete_product_random_weight_sku}
+    And Response body parameter should be:    [data][attributes][items][0][quantity]    1
+    And Response should contain the array of a certain size:    [data][attributes][items]    1
+    And Response body parameter should be:    [data][attributes][items][0][salesUnit][conversion]    1
+    And Response body parameter should be:    [data][attributes][items][0][salesUnit][precision]    100
+    And Response body parameter should be:    [data][attributes][items][0][salesUnit][productMeasurementUnit][name]    ${packaging_unit_m_name}
+    And Response body parameter should be:    [data][attributes][items][0][salesUnit][productMeasurementUnit][code]    ${packaging_unit_m}
+    And Response body parameter should be:    [data][attributes][items][0][amount]    5.0000000000
```
Comment (OKrivtsova) [inline tests/api/b2b/glue/order_endpoints/orders/positive.robot:252]: move please amount conversion and precision to variables - they will be different for different envs and will cause issues when moving to b2c
possibly we also want to move sales unit to a variable

File: tests/api/b2b/glue/order_endpoints/orders/positive.robot [hunk 9]
```diff
@@ -0,0 +1,527 @@
...
+    And Response body parameter should be:    [data][attributes][items][0][sku]    ${concrete_available_with_stock_and_never_out_of_stock}
+    And Response body parameter should be:    [data][attributes][items][0][quantity]    1
+    And Response body parameter should be:    [data][attributes][items][1][name]    ${concrete_product_with_options_name}
+    And Response body parameter should be:    [data][attributes][items][1][sku]    ${concrete_product_with_options}
+    And Response body parameter should be:    [data][attributes][items][1][quantity]    1
+    And Response body parameter should be:    [data][attributes][items][2][name]    ${concrete_product_with_options_name}
+    And Response body parameter should be:    [data][attributes][items][2][sku]    ${concrete_product_with_options}
+    And Response body parameter should be:    [data][attributes][items][2][quantity]    1
+
+Get_order_by_order_id_with_item_&_quantity>10
```
Comment (OKrivtsova) [inline tests/api/b2b/glue/order_endpoints/orders/positive.robot:283]: please rename the test to Get_order_by_order_id_with_nonsplit_item

Comment (vadym-uskov) [reply tests/api/b2b/glue/order_endpoints/orders/positive.robot:283]: fixed

File: tests/api/b2b/glue/order_endpoints/orders/positive.robot (hunk 8 above)
Comment (vadym-uskov) [reply tests/api/b2b/glue/order_endpoints/orders/positive.robot]: added variables:
"concrete_product_random_weight_amount": "5.0000000000"
"concrete_product_random_weight_conversion": "1"
"concrete_product_random_weight_precision": "100"

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**